python-sdk/src/astro/files/types/csv.py

```diff
@@ -15,14 +16,21 @@ class CSVFileType(FileType):
 
     # We need skipcq because it's a method overloading so we don't want to make it a static method
     def export_to_dataframe(
-        self, stream, columns_names_capitalization="original", **kwargs
+        self,
+        stream,
+        pandas_options: PandasLoadOptions | None = None,
+        columns_names_capitalization="original",
+        **kwargs,
     ) -> pd.DataFrame:  # skipcq PYL-R0201
         """read csv file from one of the supported locations and return dataframe
 
         :param stream: file stream object
+        :param pandas_options: Pandas option to pass to the Pandas lib while reading csv
         :param columns_names_capitalization: determines whether to convert all columns to lowercase/uppercase
             in the resulting dataframe
         """
+        if pandas_options is not None:
+            kwargs.update(pandas_options.to_dict)
```

Comment:
Can you explain why are adding these to the kwargs? Is it safe to pass arbitrary kwargs to it?

Reply:
The kwargs here are not exposed to the public interface here (i.e in operator, decorator) so exposing a parameter `pandas_options` in load_file which will have attributes corresponding to pandas `read_csv` etc method and updating the kwargs with these values. https://github.com/astronomer/astro-sdk/pull/1466/files#diff-8927674ffbb5764b3494eb1634934aa428126e76fe1f43479de1b4cfae0b5334